cmd/zsysd/completion.go

```diff
@@ -11,18 +11,52 @@ import (
 func installCompletionCmd(rootCmd *cobra.Command) {
 	prog := rootCmd.Name()
 	var completionCmd = &cobra.Command{
-		Use:   "completion",
+		Use:   "completion [bash|zsh|powershell]",
 		Short: i18n.G("Generates bash completion scripts"),
-		Long: fmt.Sprintf(i18n.G(`To load completion run
+		Long: fmt.Sprintf(i18n.G(`To load completions:
 
-. <(%s completion)
+Bash:
 
-To configure your bash shell to load completions for each session add to your ~/.bashrc or ~/.profile:
+  $ source <(yourprogram completion bash)
 
-. <(%s completion)
+  # To load completions for each session, execute once:
+  # Linux:
+  $ yourprogram completion bash > /etc/bash_completion.d/yourprogram
+  # macOS:
+  $ yourprogram completion bash > /usr/local/etc/bash_completion.d/yourprogram
+
+Zsh:
+
+  # If shell completion is not already enabled in your environment,
+  # you will need to enable it.  You can execute the following once:
+
+  $ echo "autoload -U compinit; compinit" >> ~/.zshrc
+
+  # To load completions for each session, execute once:
+  $ yourprogram completion zsh > "${fpath[1]}/_yourprogram"
+
+  # You will need to start a new shell for this setup to take effect.
+
+PowerShell:
+
+  PS> yourprogram completion powershell | Out-String | Invoke-Expression
+
+  # To load completions for every new session, run:
+  PS> yourprogram completion powershell > yourprogram.ps1
+  # and source this file from your PowerShell profile.
 `), prog, prog),
+		DisableFlagsInUseLine: true,
+		ValidArgs:             []string{"bash", "zsh", "fish", "powershell"},
+		Args:                  cobra.ExactValidArgs(1),
```

Comment:
What do you think about the following:
- Have 0 or 1 arg
- If 0 arg -> detect current shell and execute the expected completion
- If 1 arg: ensuring (as with corba.ExactValidArgs that you have one of the whitelisted list)
- Based on the Shell detected, (I think you can do that with a PreRun), modify completionCmd.Long message to include one of the 3 snippets (bash, zsh or powershell). If not found, prints all 3.

As a sidenote, you added "fish" which is not handled.